app/favorites/views.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+"""Handle favorites views.
+
+Copyright (C) 2019 Gitcoin Core
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+import json
+from fnmatch import filter
+
+from django.http import JsonResponse
+
+from dashboard.models import Bounty, Profile, User
+from dashboard.router import BountySerializer, ProfileSerializer
+from grants.models import Grant
+from grants.router import GrantSerializer
+from kudos.models import Token
+from kudos.router import TokenSerializer
+from rest_framework import status
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import Favorite
+
+
+@api_view(['POST', 'DELETE'])
+@authentication_classes((SessionAuthentication, BasicAuthentication))
```

Comment:
@kuhnchris I don't think we are sharing basicauth and sessinauth with the front end just a http cookies, that means there is not possible to get http cookies from javascript. 
That is the way if you do a jwt auth or something like that and store the tokens on login to use later in any auth request.

Reply:
unless is taking from the request tokens, I think that is the case, anyway get works perfectly but post give me 403

Reply:
![image](https://user-images.githubusercontent.com/1358093/63058411-06977480-bec3-11e9-8c72-f4b011fc8677.png)

Reply:
you can test it on the console with something like:
```
var postFav = fetchData('api/v0.1/favorites/user/2', 'POST')
$.when(postFav).then((response, status, statusCode) => {
  console.log(response, status, statusCode)
})
```

Reply:
Well if you are authorized (e.a. you have access to admin) then it does indeed work. (as i tried it myself) - I'm not 100% sure how we handle the github auth - do we have any other paths where we require authentification where I could steal the code from? Else we may need to check how to protect the data. (I tested it via Boomerang REST Extension for Chrome)

Reply:
Also, why would I use Javascript for auth? that's a pyhton file you are looking at - it checks on the backend (django) if you are auth - and if you are authed on the entire app it checks the User Object's permission and executes the function if the permission is sufficient. I do not think there are any additional API requirements, but we'd need to check RESTful API's documentation (django-rest-framework) if they may impose a special permissions...

Reply:
After a quick check - not sure but maybe I can use the django.contrib.auth.decorators.login_required.

It SHOULD provide a request.user if it does, and a request.user.profile (a gitcoin Profile object) if it is authenticated.

I can try that out if it's fine with you.

See dashboard/views.py and dashboard/utils.py - helper_handle_access_token and new_bounty method, those are taking those assumtions.

```
from django.contrib.auth.decorators import login_required
...
@login_required
def new_bounty(request):
...
        user=request.user if request.user.is_authenticated else None,
...
```

Reply:
OK I got what you mean now - i do get a 403, but the 403 tells me I need the CSRF token set. We do not have access to the CSRF Token tho. This only happens if we use SessionAuth. Which we do. So we either need to CSRFExempt the API calls (apparently not really viable, or easy doable), or use a different auth scheme like tokens, as you mentioned before.

Reply:
yes but anyway try to comment the auth part and do a post with the command I sent to you.  Get are working with the same auth validator but this endpoint with Post doesn't work also without that

Reply:
@kuhnchris how is going? any problem? Ping me if you have any question about my comment.

Reply:
well, all my attempts to implement this right now failed, from trying to bypass csrf to try to use another scheme (which we cannot) to trying to bypass the csrf somehow again. up to now success until now. any ideas/pointers? The only "idea" i could get at this point is to redo everything in a token-auth based thing but that'd break explorer and co. :-(